File: backend-springboot/src/main/java/com/javeriana/pontimovil/ponti_movil/dto/gestion_rutas/ruta_recibida/rRutaRecibidaDTO.java
package com.javeriana.pontimovil.ponti_movil.dto.gestion_rutas.ruta_recibida;

import lombok.Getter;
import lombok.Setter;
import java.io.Serializable;
import lombok.*;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@EqualsAndHashCode
@NoArgsConstructor
public class rRutaRecibidaDTO implements Serializable {
    UUID id;
    String codigo;
    rHorarioDTO horario;
    List<String> diasSemana;
    List<rEstacionRecibidaDTO> estaciones;
    List<rBusDTO> buses;
    List<rConductorDTO> conductores;
}

File: frontend-angular/src/app/dto/gestionar-rutas/ruta-recibida/r-ruta-recibida-dto.ts
import {RHorarioDto} from "./r-horario-dto";
import {RBusDto} from "./r-bus-dto";
import {RConductorDto} from "./r-conductor-dto";
import {REstacionRecibidaDto} from './r-estacion-recibida-dto';

export class RRutaRecibidaDto {
  constructor(
    public id: string,
    public codigo: string,
    public horario: RHorarioDto,
    public diasSemana: string[],
    public estaciones: REstacionRecibidaDto[],
    public buses: RBusDto[],
    public conductores: RConductorDto[]
  ) {}
}
